ros/src/computing/perception/detection/lidar_tracker/packages/lidar_imm_ukf_pda_track/nodes/imm_ukf_pda/ukf.cpp

```diff
@@ -324,20 +408,288 @@ void UKF::predictionIMMUKF(const double dt)
   /*****************************************************************************
   *  Prediction
   ****************************************************************************/
-  prediction(dt, 0);
-  prediction(dt, 1);
-  prediction(dt, 2);
-
+  prediction(dt, MotionModel::CV);
+  prediction(dt, MotionModel::CTRV);
+  prediction(dt, MotionModel::RM);
   /*****************************************************************************
   *  Update
   ****************************************************************************/
-  updateLidar(0);
-  updateLidar(1);
-  updateLidar(2);
+  updateLidar(MotionModel::CV);
+  updateLidar(MotionModel::CTRV);
+  updateLidar(MotionModel::RM);
+}
+
+void UKF::findMaxZandS(Eigen::VectorXd& max_det_z, Eigen::MatrixXd& max_det_s)
+{
+  double cv_det = s_cv_.determinant();
+  double ctrv_det = s_ctrv_.determinant();
+  double rm_det = s_rm_.determinant();
+
+  if (cv_det > ctrv_det)
+  {
+    if (cv_det > rm_det)
+    {
+      max_det_z = z_pred_cv_;
+      max_det_s = s_cv_;
+    }
+    else
+    {
+      max_det_z = z_pred_rm_;
+      max_det_s = s_rm_;
+    }
+  }
+  else
+  {
+    if (ctrv_det > rm_det)
+    {
+      max_det_z = z_pred_ctrv_;
+      max_det_s = s_ctrv_;
+    }
+    else
+    {
+      max_det_z = z_pred_rm_;
+      max_det_s = s_rm_;
+    }
+  }
+}
+
+void UKF::updateEachMotion(const double detection_probability, const double gate_probability, const double gating_thres,
```

Comment:
In this function, it seems that almost same caliculations are made for cv, ctrv, and rm. Making a function might make this function simple.

Reply:
This PR aims to change topic type.
Definitely modifying in next PR.